### PR TITLE
Tag FunctionalCollections v0.2.0, Patchwork v0.3.0, Escher v0.3.3, and ComposeDiff v0.0.2

### PR DESCRIPTION
**Pull request by shashi**

### CODE REVIEW

**Comment by tkelman** [comment]
Escher's Travis is failing and way out of date.

File: Escher/versions/0.3.3/requires [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ArgParse
+Colors
+Compat
+Dates
+JSON
+Markdown
+Mux 0.1.0
+Reactive 0.3.0
+Requires
+Patchwork 0.2.3
```
Comment (tkelman) [inline Escher/versions/0.3.3/requires]: actually 0.3?

File: Escher/versions/0.3.3/requires [hunk 2]
```diff
@@ -0,0 +1,13 @@
+julia 0.4
+ArgParse
+Colors
+Compat
```
Comment (tkelman) [inline Escher/versions/0.3.3/requires]: what minimum version?

**Comment by tkelman** [comment]
Does Escher pass locally? Looks like the failure is because this new tag of Patchwork isn't merged yet.

**Comment by shashi** [comment]
@tkelman it does. It fails only the first time on 0.5 if I clear `~/.julia/lib/v0.5/*`.

**Comment by tkelman** [comment]
Sounds like it's related to the unreliability of most optional dependency mechanisms in combination with precompilation. These things should really be moved to separate packages, or disabled entirely until there's a supported mechanism in the language for it.

Is the first-time failure new with this tag, or did it exist for the previous version too?

**Comment by shashi** [comment]
previous version crashed for a different reason on 0.5 and nightly. This is an improvement.

**Pull request merged**